domain/schema/model/sql/0024-relation.sql

```diff
@@ -284,7 +284,15 @@ SELECT
 FROM relation_endpoint AS re
 JOIN application_endpoint AS ae ON re.endpoint_uuid = ae.uuid
 JOIN charm_relation AS cr ON ae.charm_relation_uuid = cr.uuid
-JOIN application AS a ON ae.application_uuid = a.uuid;
+JOIN application AS a ON ae.application_uuid = a.uuid
+ORDER BY application_name;
+
+CREATE VIEW v_relation_key AS
+SELECT
+    relation_uuid,
+    group_concat(concat(application_name, ':', endpoint_name), ' ') AS relation_key
```

Comment:
A relation key exists for the relation tag and was never intended to be saved in the db. The relation uuid is a better identification of a relation.

It is also more complex than the query putting the application and and endpoint together: https://github.com/juju/juju/blob/acd758f56a0374629264489a2e83e96f2f56c0c1/core/relation/key.go#L47-L69
I'm concerned that we'll have multiple "keys" which don't match easily.  Exhibited by the changes to model migration tested.

Reply:
This is now reverted - explicit logic is used to resolve relation uuids to keys and visa versa, rather than using a db view.